tests/test_tools.py

```diff
@@ -125,6 +125,19 @@ def example_fxn(x: int, y: str, z: float) -> None:
     assert isinstance(z, float)
 
 
+def state_out_of_order(state: dict, x: int = 0) -> None:
```

Comment:
[nit] "out of order" can leave one wondering, maybe name `state_not_final_arg` to be more explicit